src/v/kafka/server/group.cc

```diff
@@ -1736,6 +1738,40 @@ group::store_txn_offsets(txn_offset_commit_request r) {
     co_return txn_offset_commit_response(r, error_code::none);
 }
 
+kafka::error_code map_store_offset_error_code(std::error_code ec) {
+    if (ec.category() == raft::errc_category()) {
+        switch (raft::errc(ec.value())) {
+        case raft::errc::success:
+            return error_code::none;
+        case raft::errc::timeout:
+        case raft::errc::shutting_down:
+            return error_code::request_timed_out;
+        case raft::errc::not_leader:
+            return error_code::not_coordinator;
+        case raft::errc::disconnected_endpoint:
+        case raft::errc::exponential_backoff:
+        case raft::errc::non_majority_replication:
+        case raft::errc::vote_dispatch_error:
+        case raft::errc::append_entries_dispatch_error:
+        case raft::errc::replicated_entry_truncated:
+        case raft::errc::leader_flush_failed:
+        case raft::errc::leader_append_failed:
+        case raft::errc::configuration_change_in_progress:
+        case raft::errc::node_does_not_exists:
+        case raft::errc::leadership_transfer_in_progress:
+        case raft::errc::node_already_exists:
+        case raft::errc::invalid_configuration_update:
+        case raft::errc::not_voter:
+        case raft::errc::invalid_target_node:
+        case raft::errc::replicate_batcher_cache_error:
+        case raft::errc::group_not_exists:
```

Comment:
Should this be group_id_not_found?

Reply:
the four comments that @twmb left are resolved without any new code or replies. did some out-of-band communication happen that needs to be documented here?

Reply:
The out of band communication is that these errors are raft errors that will not actually be hit here, but are handled for the sake of comprehensive enum compilation validation. A followup PR will separate these errors into two blocks: errors that we know are not bubbled to clients, and those that can be.

Reply:
thanks for adding this context to the discussion @twmb